internal/cluster/cluster.go

```diff
@@ -522,32 +524,52 @@ func (m *Manager) ClusterMonitoring() {
 		if len(clusters) == 0 {
 			break
 		}
-		for _, cluster := range clusters {
-			if !m.leaderElector.IsLeader() {
-				m.log.Debugf("Not a leader, exiting ClusterMonitoring")
-				return
-			}
-			if !m.SkipMonitoring(cluster) {
 
-				if err = m.setConnectivityMajorityGroupsForClusterInternal(cluster, m.db); err != nil {
-					log.WithError(err).Error("failed to set majority group for clusters")
-				}
-				clusterAfterRefresh, err = m.refreshStatusInternal(ctx, cluster, m.db)
-				if err != nil {
-					log.WithError(err).Errorf("failed to refresh cluster %s state", cluster.ID)
-					continue
-				}
-
-				if swag.StringValue(clusterAfterRefresh.Status) != swag.StringValue(cluster.Status) {
-					log.Infof("cluster %s updated status from %s to %s via monitor", cluster.ID,
-						swag.StringValue(cluster.Status), swag.StringValue(clusterAfterRefresh.Status))
-				}
+		jobs := make(chan *common.Cluster, limit)
+		for i := range clusters {
+			jobs <- clusters[i]
+		}
 
-				if m.shouldTriggerLeaseTimeoutEvent(cluster, curMonitorInvokedAt) {
-					m.triggerLeaseTimeoutEvent(ctx, cluster)
+		var wg sync.WaitGroup
+		for i := 0; i < m.MonitorWorkersNum; i++ {
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+				for {
+					select {
```

Comment:
Maybe iteration over channel will be more suitable https://gobyexample.com/range-over-channels

Reply:
It's possible, not sure if better, it will change the design a bit, to stop the iteration it will need to close the channel. for that, I will need another channel that will collect the results in order to wait for the jobs to complete in order to run the next batch. 
Do you think that it will be better?

Reply:
Seems more elegant to me.  With combination of running workers per cycle no batch, then you will create channel in the beginning of the monitoring loop, and close it in the end.  
It relates to the comment that I think that using the default clause in the select is incorrect.

Reply:
Done